examples/00-load/create-sphere.py

```diff
@@ -0,0 +1,261 @@
+# Here, we will used the convention that ``theta`` is the
+# azimuthal angle, similar to longitude on the globe.  ``phi`` is the
+# polar angle, similar to latitude on the globe.
+
+radius = 0.5
+ntheta = 9
+nphi = 12
+theta = np.linspace(0, 2 * np.pi, ntheta)
+phi = np.linspace(0, np.pi, nphi)
+
+r_, phi_, theta_ = np.meshgrid([radius], phi, theta, indexing='ij')
+x, y, z = pv.spherical_to_cartesian(r_, phi_, theta_)
+mesh = pv.StructuredGrid(x, y, z)
+
+# %%
+# The mesh has :attr:`~pyvista.CellType.QUAD` cells. cells. The cells that look triangular
+# at the poles are actually degenerate quadrilaterals, i.e. two
+# points are coincident at the pole, as will be shown later.
+
+mesh.plot(show_edges=True)
+
+# %%
+# The mesh is of type :class:`pyvista.StructuredGrid`.
+
+mesh
+
+# %%
+# The first cell is at the top pole, and it is a :attr:`~pyvista.CellType.QUAD` cell.
+
+cell = mesh.get_cell(0)
+cell.type
+
+# %%
+# The first cell has two degenerate points.
+
+cell.points
+
+# %%
+# The cells on either side of the 'seam' along the start and end of
+# the azimuthal component are not connected. These can be detected by
+# extracting the boundary edges.
+
+boundaries = mesh.extract_feature_edges(
+    non_manifold_edges=True, feature_edges=False, manifold_edges=False
+)
+boundaries
+
+# %%
+# Visualize this by plotting the boundary edges of the mesh.
+
+pl = pv.Plotter()
+pl.add_mesh(mesh, show_edges=True)
+pl.add_mesh(boundaries, line_width=10, color='red')
+pl.show()
+
+# %%
+# Generate quadrilateral mesh of Sphere
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# This example shows how a more complicated mesh can be defined.
+#
+# In contrast to the example above, this example generates a mesh
+# that does not have degenerate points at the poles. :attr:`~pyvista.CellType.TRIANGLE` cells
+# will be used at the poles.  First, regenerate the structured data.
+
+radius = 0.5
+ntheta = 9
+nphi = 12
+theta = np.linspace(0, 2 * np.pi, ntheta)
+phi = np.linspace(0, np.pi, nphi)
+
+# %%
+# We do not want duplicate points, so remove the duplicate in theta, which
+# results in 8 unique points in theta. Similarly, the poles at ``phi=0`` and
+# ``phi=pi`` will be handled separately to avoid duplicate points, which
+# results in 10 unique points in phi.  Remove these from the grid in spherical
+# coordinates.
+
+theta = theta[:-1]
+ntheta -= 1
+phi = phi[1:-1]
+nphi -= 2
+
+# %%
+# Use :func:`pyvista.spherical_to_cartesian` to generate cartesian coordinates for points in the ``(N, 3)``
+# format required by PyVista.  Note that this method results in
+# the theta variable changing the fastest.
+
+r_, phi_, theta_ = np.meshgrid([radius], phi, theta, indexing='ij')
+x, y, z = pv.spherical_to_cartesian(r_, phi_, theta_)
+points = np.vstack((x.ravel(), y.ravel(), z.ravel())).transpose()
+
+# %%
+# The first and last points are the poles.
+
+points = np.insert(points, 0, [0.0, 0.0, radius])
+points = np.append(points, [0.0, 0.0, -radius])
```

Comment:
This change is needed to ensure `points` keeps its shape `(N, 3)`. Otherwise, these methods flatten the input.
```suggestion
points = np.insert(points, 0, [0.0, 0.0, radius], axis=0)
points = np.append(points, [[0.0, 0.0, -radius]], axis=0)
```

Passing a 1D flat array was previously allowed, and this flattened array was reshaped to `(-1,3)` automatically by `pyvista.vtk_points`. This was changed in https://github.com/pyvista/pyvista/pull/6876. And so, this example now fails and `vtk_points` complains that the input is not a single point or a 2D array of points.

From reading this example (and not knowing that `insert` and `append` will flatten by default), I _thought_ this code was trying to keep the array as `(N,3)`. In that case, the suggestion above is good. But, if the expected use-case for `vtk_points` is to pass it a flat array of points and have it reshape it, then we should probably modify `vtk_points` to allow this again.

My understanding of this, however, is that `vtk_points` should only allow either a _single_ point (1D array) or multiple points (2D array). Indeed, there are no tests in the code checking that flattened point arrays are acceptable input for this function.

Reply:
Your suggestion is a good one. I find the differences for `np.insert` and `np.append` to be confusing in this case, but maybe it is my mental block.  On first thought, I'm +1 for requiring points to be (N,3).  If it is flat, then it is ambiguous which order it should be reshaped.